src/components/DataTable/index.tsx

```diff
@@ -11,6 +11,7 @@ export type DataTableProps<Data extends object> = {
   data: readonly Data[]
   columns: readonly Column<Data>[]
   renderRowSubComponent?: SubComponent<Data>
+  onClickRow?: () => void
```

Comment:
I'm not sure how this will be useful if it it doesn't know which row was clicked or what data the row contained